### PR TITLE
KARAF-6680 Support OSGi R7 JSON configuration format

### PR DESCRIPTION
**Pull request by oliverlietz**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -190,6 +191,8 @@
 
         <felix.bundlerepository.version>2.0.10</felix.bundlerepository.version>
         <felix.configadmin.version>1.9.16</felix.configadmin.version>
+        <felix.converter.version>1.0.12</felix.converter.version>
+        <felix.cm.json.version>0.0.1-SNAPSHOT</felix.cm.json.version>
```
Comment (jbonofre) [inline pom.xml:195]: We would need a release here. I can tackle that.

**Comment by jbonofre** [comment]
FYI Felix CM JSon release is in vote. I will update as soon as it's in Maven Central.

**Comment by oliverlietz** [comment]
@jbonofre The first commit is for factory PIDs and names only (#1096) which is not yet merged into master. Both are related to OSGi R7 configurations but different scope. Squashing would mix it up.

**Comment by jbonofre** [comment]
@oliverlietz can you update to released version of Felix CM JSon or do you want me to do it ? Thanks !

**Comment by oliverlietz** [comment]
@jbonofre, Felix CM JSON updated to 1.0.2

**Comment by jbonofre** [comment]
Thanks. Unfortunately, the feature verify fails (the framework feature can't be resolved). Let me take a look.

**Comment by oliverlietz** [comment]
@jbonofre See [SLING-9218](https://issues.apache.org/jira/browse/SLING-9218), framework feature resolved after upgrading Sling Commons Johnzon to 1.2.2 which is still under vote. My full Karaf build is still running executing integration tests right now.

**Comment by oliverlietz** [comment]
@jbonofre [Sling Commons Johnzon 1.2.2](https://search.maven.org/artifact/org.apache.sling/org.apache.sling.commons.johnzon/1.2.2/jar) is now on Central.

**Comment by jbonofre** [comment]
retest this please

**Comment by jbonofre** [comment]
I don't think it's a good idea to add this directly in the `framework` feature, it should be optional as some users don't care about json support. I would rather create a separate feature in standard and create a service in config.
I actually have a big refactoring on the features coming, including this kind of change.

**Comment by jbonofre** [comment]
Superseeded by #1222

**Pull request closed without merging**